haskell/src/Indigo/Contracts/Transferlist/External.hs

```diff
@@ -0,0 +1,157 @@
+-- SPDX-FileCopyrightText: 2020 Globacap
+-- SPDX-License-Identifier: MPL-2.0
+--
+-- | This a dummy file that resembles lorentz-contract-safelist parameter needed
+-- for origination.
+--
+-- For more information you can refer to
+-- https://github.com/tqtezos/lorentz-contract-transferlist/tree/476c0429adf74041ed7d26c44e5e8e73f5a54f36
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Indigo.Contracts.Transferlist.External
+  ( Parameter
+  , Storage (..)
+  , TransferlistId
+  , OutboundTransferlists (..)
+  , convertToExternalStorage
+  , dummyExternalContract
+  ) where
+
+import qualified Data.List as List
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+
+import qualified Indigo.Contracts.Transferlist.Internal as Internal
+import Lorentz as L
+
+type TransferlistId = Natural
+
+data TransferParams = TransferParams
+  { from :: Address
+  , tos :: [Address]
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass IsoValue
+  deriving anyclass HasTypeAnn
+
+data UpdateUserParams = UpdateUserParams
+  { user :: Address
+  , userTransferlist :: Maybe TransferlistId
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass IsoValue
+  deriving anyclass HasTypeAnn
+
+data TransferlistOutboundParams = TransferlistOutboundParams
+  { transferlist :: TransferlistId
+  , newOutboundTransferlists :: Maybe OutboundTransferlists
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass IsoValue
+  deriving anyclass HasTypeAnn
+
+data OutboundTransferlists = OutboundTransferlists
+  { unrestricted :: Bool
+  , allowedTransferlists :: Set TransferlistId
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass IsoValue
+  deriving anyclass HasTypeAnn
+
+data AssertTransferlistParams = AssertTransferlistParams
+  { transferlistId :: Natural
+  , contents :: Maybe OutboundTransferlists
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass IsoValue
+  deriving anyclass HasTypeAnn
+
+data Parameter
+  = AssertTransfers [TransferParams]
+  | AssertReceivers [Address]
+  | OtherParameter ManagementParameter
+  deriving stock (Generic, Show)
+  deriving anyclass IsoValue
+
+instance ParameterHasEntryPoints Parameter where
+  type ParameterEntryPointsDerivation Parameter = EpdRecursive
+
+data ManagementParameter
```

Comment:
We don't want to call any entrypoints of transferlist except for `AssertTransfers` and `AssertReceivers`, so we don't need this type.